### PR TITLE
ENH/CLN: Improve diagnostics 

### PR DESCRIPTION
**Pull request by bashtage**
- [x] closes #6408
- [X] tests added / passed. 
- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in master and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/master -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

File: statsmodels/stats/_lilliefors.py [hunk 1]
```diff
@@ -220,32 +220,32 @@ def pval_lf(d_max, n):
 
 def kstest_fit(x, dist='norm', pvalmethod=None):
     """
-    Lilliefors test for normality or an exponential distribution
+    Test assumed normal or exponetntial distribution using Lilliefor's test.
```
Comment (josef-pkt) [inline statsmodels/stats/_lilliefors.py]: typo exponential

**Comment by josef-pkt** [comment]
main changes are in 09f775e5813734a1c9bdbb1b7eb834cbf145b276
changes show up because history ignored clean move

There are several changes that change behavior and have deprecations

needs review, I'm not sure I agree with all changes

I assume you are not planning to merge this for the upcoming release
assign milestone

**Comment by josef-pkt** [comment]
most changes look good, I like the cov_type in some of the tests, do they have unit tests?

I would have preferred to keep the `diagnostic` functions out of `diagnostic.py`
now it mixes api style imports, and functions that are directly included.

changes to return types need review, and (decision on what the pattern should finally be).


Do you have a reference for NOT using aic lag search in the `acorr` test?
I still don't like the removal of autolag option in there. 
(for Stata Baum has a good implementation, IIRC, but I didn't check the details. One feature I thought would be nice to have is that he has an option to test whether acorr is zero after an initial user provided lag.)

**Comment by bashtage** [comment]
> Do you have a reference for NOT using aic lag search in the `acorr` test?

It is just testing after selection.  As a rule size isn't controlled here.

**Comment by bashtage** [comment]
> changes to return types need review, and (decision on what the pattern should finally be).

There are no changes to the return types.  There are some deprecation promises about the future, but these can always be changed.

**Comment by bashtage** [comment]
I would like to get it in for 0.11 since I was super careful to not break anything (other than make some noise).

**Comment by josef-pkt** [comment]
It's not good software practice to make large, not urgent changes like this between rc and release

**Comment by bashtage** [comment]
> It's not good software practice to make large, not urgent changes like this between rc and release

I agree, but it isn't a great idea to proffer half tested code with missing documentation.  Most importantly I think the only way to get any real testing is to get it out.  Even rc testing is pretty light. Will probably need another week with an rc2 though. Around 0.5% of installs over the past month have been 0.11.0rc1.

Idx | Downloads | Version |  
-- | -- | -- | --
1 | 324 | 0.9.0rc1 |  
2 | 524408 | 0.9.0 |  
3 | 1113 | 0.8.0rc1 |  
4 | 224252 | 0.8.0 |  
5 | 24090 | 0.6.1 |  
6 | 460 | 0.6.0-rc2 |  
7 | 100 | 0.6.0-rc1 |  
8 | 3955 | 0.6.0 |  
9 | 338 | 0.5.0rc1 |  
10 | 466 | 0.5.0 |  
11 | 386 | 0.4.3 |  
12 | 388 | 0.4.1 |  
13 | 388 | 0.4.0 |  
14 | 11287 | 0.11.0rc1 |  
15 | 1497524 | 0.10.2 |  
16 | 230947 | 0.10.1 |  
17 | 9179 | 0.10.0rc2 |  
18 | 73528 | 0.10.0

**Pull request merged**